dsp/modules/hf_client.py

```diff
@@ -116,7 +116,16 @@ def send_hftgi_request_v00(arg, **kwargs):
 class HFClientVLLM(HFModel):
     def __init__(self, model, port, url="http://localhost", **kwargs):
         super().__init__(model=model, is_client=True)
-        self.url = f"{url}:{port}"
+        
+        if type(url) == list:
+            self.urls = ['/v1'.join(url_.split('/v1')[:-1]) for url_ in url]
```

Comment:
this seems a bit extra - would it be better to specify some base formatting in the documentation and having a condition check on that instead? feel free to add it!

Reply:
I agree with this. It's unclear why we need to do this splitting and joining.  I feel this might lead to unintuitive behavior, for example, if I don't include /v1/completions in my URLs now it drops the URL completely:

```
>>> urls = ["localhost:8000/v1/completions", "server.com:7410/v1/completions", "mycluster:4000", "oldservice.co:3000/v1/service/v1/completions"]
>>> new_urls = ['/v1'.join(url_.split('/v1')[:-1]) for url_ in urls]
>>> new_urls
['localhost:8000', 'server.com:7410', '', 'oldservice.co:3000/v1/service']
```

This would break the existing way people are inputting URLs if they just convert to list form.

Reply:
Also we might want to add a warning if the user passes a port and list of URLs because the port is now ignored.